internal/controllers/common/common_utils.go

```diff
@@ -77,21 +89,46 @@ func ClusterUniqueNameWithPrefix(gvk *schema.GroupVersionKind, prefix string, na
 	return ClusterUniqueNameWithPrefixTargetNS(gvk, prefix, name, namespace, "")
 }
 
+// ClusterUniqueShortName returns a name for cluster-scoped objects that is
+// uniquely identified by a namespace and name. The total length should be
+// at most 63 characters.
+func ClusterUniqueShortName(gvk *schema.GroupVersionKind, name string, namespace string) string {
+	return clusterUniqueName(gvk, "", name, namespace, "", true)
+}
+
 // ClusterUniqueNameWithPrefixTargetNS returns a name for cluster-scoped objects that is
 // uniquely identified by a namespace and name, and a target namespace.
 // Appends the prefix to the provided Kind.
 func ClusterUniqueNameWithPrefixTargetNS(gvk *schema.GroupVersionKind, prefix string, name string, namespace string,
 	targetNS string) string {
+	return clusterUniqueName(gvk, prefix, name, namespace, targetNS, false)
+}
+
+func clusterUniqueName(gvk *schema.GroupVersionKind, prefix string, name string, namespace string, targetNS string,
+	short bool) string {
 	prefixWithKind := strings.ToLower(gvk.Kind)
+	if short {
+		// Prefix is 5 characters: 4 from kind, and hyphen
+		prefixWithKind = prefixWithKind[:4] // TODO can we use a weaker and shorter hash and keep the full kind?
```

Comment:
@andrewazores what do you think about using a non-cryptographic hash function for this instead?
With https://pkg.go.dev/hash/fnv, we could use:
`cryostat-<128-bit hash>` instead of `cryo-<224-bit hash>`

Reply:
Do you mean just swapping out the hash functions, or would this also include changing the prefix before the hash or changing the hash inputs?